### DISCUSSION TITLE
Critical CSS from components inside layouts/default.vue

### DISCUSSION THREAD

**Discussion by gavin667** [nuxt/nuxt#23013] [Questions] [answered]
Hi 👋 I was hoping someone could help me understand something around layouts and css. I'm not sure if it's a bug or not. More likely a misunderstanding.

Nuxt: 3.7.0
Mode: Universal Rendering

I have a `layouts/default.vue` file that looks like:

```
//layouts/default.vue
<div>
    <PromoBanner />
    <MainNavigation />
    <div>
      <slot />
    </div>
</div>
<script setup></script>
```

Then I have a page at `pages/support/index.vue` that looks like:

```
//pages/support/index.vue
<template>
  <div>
    <p>Content here</p>
  </div>
</template>
<script></script>
<style lang="postcss">
// Page specific styles
</style>
```

When running `yarn dev` everything loads as expected.

However when running `yarn build` the CSS from the layout components `<PromoBanner/>` and `<MainNavigation/>` are put inside `_nuxt/default.c92e2931.css` and loaded after the page has rendered leading to FOUC. Any components or styles defined in `pages/support/index.vue` do have their CSS inlined correctly on initial page load avoiding FOUC on that particular section.

If I get rid of `layouts/default.vue` entirely and put the two components (`<PromoBanner />` and  `<MainNavigation />`) inside `app.vue` the css is inlined correctly in the production build.

With and without the `app.vue` file the issue is the same:

```
//app.vue
<template>
<NuxtLayout>
   <NuxtPage/>
<NuxtLayout
</template>
```

**Is there a way in Nuxt 3 to get components that are defined in a layout file to have their CSS rendered inline on initial page load?**

**Comment by logotip4ik** [comment]
You can try using `experimental.inlineSSRStyles`. It can receive a function that returns Boolean, whether inline styles or not. 

[More info on nuxt.com](https://nuxt.com/docs/api/configuration/nuxt-config#inlinessrstyles)

**Reply by gavin667** [reply]
Thanks @logotip4ik, I already had that option set. I've reproduced the issue and you can see the further details I added below if you are interested 😄

**Comment by gavin667** [comment]
I got to the bottom of the issue and it's a weird one...

I've reproduced the issue and you can see it here - https://nuxt-3-layout-css.vercel.app and the repo I've made available here - https://github.com/gavin667/nuxt-3-layout-css

Notice how in the example the text flashes from black to blue shortly after loading? The styles have not been inlined for this component in the layout on load.

It seems to be that if I call a Pinia store in a layout file and reference a lazy loaded component that also references the same store it fails to inline the styles correctly on initial page load.

Check [the repo I made](https://github.com/gavin667/nuxt-3-layout-css) but as a simplified version:

```
// layouts/default.vue
<template>
  <div>
    <div>This is text directly on the layout.</div>

    <!-- CSS inside this is not inlined and leads to FOUC -->
    <AnotherComponent/>

    <!-- This component also references const exampleStore = useExampleStore() -->
    <LazyTestComponent v-if="showLazyComponent"/>

    <div>
        <slot />
    </div>
  </div>
</template>
<script setup>
  const exampleStore = useExampleStore()
</script>
```

```
// components/TestComponent.vue
<template>
  <div class="test-component">
    <h1>This should be red on load</h1>
  </div>
</template>

<script setup>
  const exampleStore = useExampleStore()
</script>

<style lang="postcss">
  .test-component {
    color: red;
    font-size: 1.2rem;
  }
</style>  
```

```
// components/AnotherComponent.vue
<template>
    <h1 class="another-component">
        This should be blue on load
    </h1>
</template>
<style lang="postcss">
  // these styles are missing onload
  .another-component {
    color: blue;
  }
</style>
```

**Comment by gavin667** [comment, marked as answer]
This is a bug that is being tracked in https://github.com/nuxt/nuxt/issues/22507
